commit 8e42fc6811ba94a01e194d46572c3cf803b3aae0
Author: Mike FABIAN <mfabian@redhat.com>
Date:   Tue Sep 17 17:20:32 2019 +0200

    Sync "language", "lang_name", "territory", "country_name" with CLDR/langtable
    
    Sync these values with CLDR and langtable as much as possible.  Add
    missing values.
    
    If possible, take the values from CLDR, if CLDR does not have it,
    take it from langtable. The values from langtable which are not from
    CLDR are from  Wikipedia or native speakers.

diff --git a/localedata/locales/az_AZ b/localedata/locales/az_AZ
index 6fe8839f25cc7450..3c0db0be3de1e6d3 100644
--- a/localedata/locales/az_AZ
+++ b/localedata/locales/az_AZ
@@ -243,7 +243,7 @@ country_ab3 "AZE"
 country_num 031
 country_car    "AZ"
 % Azərbaycanca
-lang_name    "az<U0259>rbaycan dili"
+lang_name    "az<U0259>rbaycan"
 lang_ab      "az"
 lang_term    "aze"
 lang_lib    "aze"
diff --git a/localedata/locales/be_BY@latin b/localedata/locales/be_BY@latin
index 43b13cc27812aaa0..fef339f6247a0eb6 100644
--- a/localedata/locales/be_BY@latin
+++ b/localedata/locales/be_BY@latin
@@ -156,7 +156,7 @@ country_ab2 "BY"
 country_ab3 "BLR"
 country_num 112
 country_car    "BY"
-lang_name    "bie<U0142>aruskaja mova"
+lang_name    "bie<U0142>aruskaja"
 lang_ab      "be"
 lang_term    "bel"
 lang_lib    "bel"
diff --git a/localedata/locales/ber_DZ b/localedata/locales/ber_DZ
index 79f3d289b1ee771c..4fa1bdfff94b9efd 100644
--- a/localedata/locales/ber_DZ
+++ b/localedata/locales/ber_DZ
@@ -8,7 +8,7 @@ escape_char /
 % exempt you from the conditions of the license if your use would
 % otherwise be governed by that license.
 
-% Amazigh Language Locale for Algeria (latin)
+% Berber Language Locale for Algeria (latin)
 % Source:
 % Contact: Pablo Saratxaga
 % Email: <pablo@mandrakesoft.com>
@@ -20,14 +20,14 @@ escape_char /
 % Users: general
 
 LC_IDENTIFICATION
-title "Amazigh language locale for Algeria (latin)"
+title "Berber language locale for Algeria (latin)"
 source ""
 address ""
 contact "Pablo Saratxaga"
 email "pablo@mandrakesoft.com"
 tel ""
 fax ""
-language "Amazigh"
+language "Berber"
 territory "Algeria"
 revision "0.1"
 date "2002-04-16"
@@ -253,14 +253,15 @@ LC_ADDRESS
 % This is the ISO_IEC TR14652 Locale definition for the
 % LC_ADDRESS
 postal_fmt  "%z%c%T%s%b%e%r"
+country_name "Lezzayer"
 %country_post ""
 country_ab2 "DZ"
 country_ab3 "DZA"
 country_num 012
 %country_isbn ""
 country_car  "DZ"
-% ⵜⴰⵎⴰⵣⵉⵖⵜ
-lang_name "<U2D5C><U2D30><U2D4E><U2D30><U2D49><U2D56><U2D5C>"
+% Tamaziɣt
+lang_name "Tamazi<U0263>t"
 %lang_ab
 lang_term "ber"
 lang_lib "ber"
diff --git a/localedata/locales/ber_MA b/localedata/locales/ber_MA
index b9bd64868cda3412..5d19354ad65069eb 100644
--- a/localedata/locales/ber_MA
+++ b/localedata/locales/ber_MA
@@ -8,7 +8,7 @@ escape_char /
 % exempt you from the conditions of the license if your use would
 % otherwise be governed by that license.
 
-% Amazigh Language Locale for Morocco (tifinagh)
+% Berber Language Locale for Morocco (tifinagh)
 % Source:
 % Contact: Pablo Saratxaga
 % Email: <pablo@mandrakesoft.com>
@@ -20,14 +20,14 @@ escape_char /
 % Users: general
 
 LC_IDENTIFICATION
-title "Amazigh language locale for Morocco (tifinagh)"
+title "Berber language locale for Morocco (tifinagh)"
 source ""
 address ""
 contact "Pablo Saratxaga"
 email "pablo@mandrakesoft.com"
 tel ""
 fax ""
-language "Amazigh"
+language "Berber"
 territory "Morocco"
 revision "0.1"
 date "2002-06-26"
@@ -204,6 +204,8 @@ LC_ADDRESS
 % This is the ISO_IEC TR14652 Locale definition for the
 % LC_ADDRESS
 postal_fmt  "%z%c%T%s%b%e%r"
+% https://en.wikipedia.org/wiki/Morocco ⵜⴰⴳⵍⴷⵉⵜ ⵏ ⵍⵎⵖⵔⵉⴱ
+country_name "<U2D5C><U2D30><U2D33><U2D4D><U2D37><U2D49><U2D5C> <U2D4F> <U2D4D><U2D4E><U2D56><U2D54><U2D49><U2D31>"
 %country_post ""
 country_ab2 "MA"
 country_ab3 "MAR"
@@ -211,7 +213,7 @@ country_num 504
 %country_isbn ""
 country_car  "MA"
 % ⵜⴰⵎⴰⵣⵉⵖⵜ
-lang_name "<U2D5C><U2D30><U2D4E><U2D30><U2D49><U2D56><U2D5C>"
+lang_name "<U2D5C><U2D30><U2D4E><U2D30><U2D63><U2D49><U2D56><U2D5C>"
 % lang_ab
 lang_term "ber"
 lang_lib "ber"
diff --git a/localedata/locales/bhb_IN b/localedata/locales/bhb_IN
index 549974de2cfd4843..80d590d068582de6 100644
--- a/localedata/locales/bhb_IN
+++ b/localedata/locales/bhb_IN
@@ -145,11 +145,13 @@ LC_ADDRESS
 % This is the ISO_IEC TR14652 Locale definition for the LC_ADDRESS category
 % generated by IBM Basic CountryPack Transformer.
 postal_fmt "%z%c%T%s%b%e%r"
+country_name "<U092D><U093E><U0930><U0924>"
 country_ab2 "IN"
 country_ab3 "IND"
 country_car "IND"
 country_num 356
-lang_name     "bhili"
+% भीली
+lang_name     "<U092D><U0940><U0932><U0940>"
 lang_ab       ""
 lang_term     "bhb"
 lang_lib      "bhb"
diff --git a/localedata/locales/bho_IN b/localedata/locales/bho_IN
index 99a91c2553ecdeae..4a0f25d6aceaa023 100644
--- a/localedata/locales/bho_IN
+++ b/localedata/locales/bho_IN
@@ -147,6 +147,7 @@ END LC_NAME
 
 LC_ADDRESS
 postal_fmt  "%z%c%T%s%b%e%r"
+country_name "<U092D><U093E><U0930><U0924>"
 country_ab2 "IN"
 country_ab3 "IND"
 country_num 356
diff --git a/localedata/locales/chr_US b/localedata/locales/chr_US
index 290d1b24551dd3ba..86868c02e9ae2e1c 100644
--- a/localedata/locales/chr_US
+++ b/localedata/locales/chr_US
@@ -113,7 +113,7 @@ END LC_NAME
 
 LC_ADDRESS
 postal_fmt "%a%N%f%N%d%N%b%N%h %s %e %r%N%T, %S %z%N%c%N"
-country_name "<U13A0><U13B9><U13F0><U13DF>"
+country_name "<U13CC><U13CA><U0020><U13A2><U13F3><U13BE><U13B5><U13CD><U13D4><U13C5><U0020><U13CD><U13A6><U13DA><U13A9>"
 country_post "USA"
 country_ab2  "US"
 country_ab3  "USA"
diff --git a/localedata/locales/crh_UA b/localedata/locales/crh_UA
index b2ac8c66d78521b0..86281a27db1cc64d 100644
--- a/localedata/locales/crh_UA
+++ b/localedata/locales/crh_UA
@@ -198,8 +198,8 @@ country_ab2 "UA"
 country_ab3 "UKR"
 country_num 804
 country_car  "UA"
-% Qırımtatarca
-lang_name    "Q<U0131>r<U0131>mtatarca"
+% qırımtatar tili
+lang_name    "q<U0131>r<U0131>mtatar tili"
 lang_term    "crh"
 lang_lib    "crh"
 END LC_ADDRESS
diff --git a/localedata/locales/csb_PL b/localedata/locales/csb_PL
index c8572f8fc54853b3..662dcefa15a2cae3 100644
--- a/localedata/locales/csb_PL
+++ b/localedata/locales/csb_PL
@@ -210,6 +210,7 @@ END LC_NAME
 
 LC_ADDRESS
 postal_fmt    "%f%N%a%N%d%N%b%N%s %h %e %r%N%z %T%N%c%N"
+country_name "P<U00F2>lsk<U00F4>"
 country_ab2 "PL"
 country_ab3 "POL"
 country_num 616
diff --git a/localedata/locales/doi_IN b/localedata/locales/doi_IN
index 0428ae50a9bf60e8..df9dfc6e2a428abb 100644
--- a/localedata/locales/doi_IN
+++ b/localedata/locales/doi_IN
@@ -156,6 +156,7 @@ END LC_NAME
 
 LC_ADDRESS
 postal_fmt  "%z%c%T%s%b%e%r"
+country_name "<U092D><U093E><U0930><U0924>"
 country_ab2 "IN"
 country_ab3 "IND"
 country_num 356
diff --git a/localedata/locales/dv_MV b/localedata/locales/dv_MV
index 0d7842f39f7edf85..ab78fa5ce33cfe0e 100644
--- a/localedata/locales/dv_MV
+++ b/localedata/locales/dv_MV
@@ -182,9 +182,8 @@ country_ab2 "MV"
 country_ab3 "MDV"
 country_num 462
 country_car "MV"
-% lang_name FIXME
-% Cannot represent Dhivehi in Thaana script
-% http://en.wikipedia.org/wiki/Maldivian_language
+% ދިވެހި
+lang_name    "<U078B><U07A8><U0788><U07AC><U0780><U07A8>"
 lang_ab      "dv"
 lang_term    "div"
 lang_lib    "div"
diff --git a/localedata/locales/eo b/localedata/locales/eo
index 33a81033e23caadd..b2ad5758004b67d8 100644
--- a/localedata/locales/eo
+++ b/localedata/locales/eo
@@ -206,7 +206,7 @@ END LC_NAME
 
 LC_ADDRESS
 postal_fmt   "%f%N%a%N%d%N%b%N%s %h %e %r%N%%z %T%N%c%N"
-lang_name    "Esperanto"
+lang_name    "esperanto"
 lang_ab      "eo"
 lang_term    "epo"
 END LC_ADDRESS
diff --git a/localedata/locales/hak_TW b/localedata/locales/hak_TW
index f62937e91b634416..87f2a00489d1574e 100644
--- a/localedata/locales/hak_TW
+++ b/localedata/locales/hak_TW
@@ -187,8 +187,8 @@ country_ab3  "TWN"
 country_num  158
 country_car "RC"
 country_isbn 957
-% 漢語客家語
-lang_name    "<U6F22><U8A9E><U5BA2><U5BB6><U8A9E>"
+% 客家語
+lang_name    "<U5BA2><U5BB6><U8A71>"
 lang_term    "hak"
 lang_lib     "hak"
 END LC_ADDRESS
diff --git a/localedata/locales/hif_FJ b/localedata/locales/hif_FJ
index 5433bb4a2a54636e..a6868b791025c8f4 100644
--- a/localedata/locales/hif_FJ
+++ b/localedata/locales/hif_FJ
@@ -176,7 +176,8 @@ country_car  "FJI"
 country_num 242
 % https://en.wikipedia.org/wiki/ISO_3166-1_numeric
 % https://en.wikipedia.org/wiki/Fiji_Hindi
-lang_name    "Fiji Baat"
+% फ़िजी बात
+lang_name    "<U0939><U093F><U0928><U094D><U0926><U0940>"
 % iso-639-1
 lang_ab      ""
 % iso-639-3
diff --git a/localedata/locales/hne_IN b/localedata/locales/hne_IN
index 7abec636304b9e67..d8b83610165a6844 100644
--- a/localedata/locales/hne_IN
+++ b/localedata/locales/hne_IN
@@ -154,6 +154,7 @@ END LC_NAME
 
 LC_ADDRESS
 postal_fmt  "%z%c%T%s%b%e%r"
+country_name "<U092D><U093E><U0930><U0924>"
 country_ab2 "IN"
 country_ab3 "IND"
 country_num 356
diff --git a/localedata/locales/ia_FR b/localedata/locales/ia_FR
index f3fbf213a3f7f267..4ce7c7fe4a416541 100644
--- a/localedata/locales/ia_FR
+++ b/localedata/locales/ia_FR
@@ -127,7 +127,7 @@ country_ab3 "FRA"
 country_num 250
 country_isbn "979-10"
 country_car "F"
-lang_name "Interlingua"
+lang_name "interlingua"
 lang_ab      "ia"
 lang_term    "ina"
 lang_lib     "ina"
diff --git a/localedata/locales/id_ID b/localedata/locales/id_ID
index 3ddd8d07daed6697..92ff81df23b059a0 100644
--- a/localedata/locales/id_ID
+++ b/localedata/locales/id_ID
@@ -155,7 +155,7 @@ country_ab2 "ID"
 country_ab3 "IDN"
 country_num 360
 country_car    "RI"
-lang_name    "Bahasa Indonesia"
+lang_name    "Indonesia"
 lang_ab      "id"
 lang_term    "ind"
 lang_lib    "ind"
diff --git a/localedata/locales/ig_NG b/localedata/locales/ig_NG
index bddd2ccde523dca8..113294a0f58ab576 100644
--- a/localedata/locales/ig_NG
+++ b/localedata/locales/ig_NG
@@ -321,11 +321,11 @@ LC_ADDRESS
 % "end of line
 postal_fmt    "%f%N%a%N%d%N%b%N%s %h %e %r%N%z %T%N%c%N"
 
-% Country name in Igbo - "Nigeria"
-country_name  "Nigeria"
+% Country name in Igbo - "Naịjịrịa"
+country_name  "Na<U1ECB>j<U1ECB>r<U1ECB>a"
 
-% Language name in Igbo - "Igbo"
-lang_name     "Igbo"
+% Language name in Igbo - "Asụsụ Igbo"
+lang_name     "As<U1EE5>s<U1EE5> Igbo"
 
 % CEPT MAILCODES are suggested
 % Alternatively use the code found on your countries postal item tracking number
diff --git a/localedata/locales/kab_DZ b/localedata/locales/kab_DZ
index a165f53f01370c94..ef572357c22615f7 100644
--- a/localedata/locales/kab_DZ
+++ b/localedata/locales/kab_DZ
@@ -154,7 +154,7 @@ END LC_NAME
 
 LC_ADDRESS
 postal_fmt   "%f%N%a%N%d%N%b%N%s %h %e %r%N%z %T%N%c%N"
-country_name "Zzayer"
+country_name "Lezzayer"
 country_ab2  "DZ"
 country_ab3  "DZA"
 country_num  012
diff --git a/localedata/locales/ks_IN b/localedata/locales/ks_IN
index 6992a5536887d9f6..e1efea342d4b55a0 100644
--- a/localedata/locales/ks_IN
+++ b/localedata/locales/ks_IN
@@ -167,7 +167,7 @@ LC_ADDRESS
 % This is the ISO_IEC TR14652 Locale definition for the LC_ADDRESS category
 % generated by IBM Basic CountryPack Transformer.
 postal_fmt  "%z%c%T%s%b%e%r"
-country_name "<U06C1><U0650><U0646><U065B><U062F><U0648><U0633><U062A><U0627><U0646>"
+country_name "<U06C1><U0650><U0646><U062F><U0648><U0633><U062A><U0627><U0646>"
 country_ab2 "IN"
 country_ab3 "IND"
 country_num 356
diff --git a/localedata/locales/ku_TR b/localedata/locales/ku_TR
index 595cdb14bc9ef29d..f6a8925608ea046d 100644
--- a/localedata/locales/ku_TR
+++ b/localedata/locales/ku_TR
@@ -182,14 +182,14 @@ END LC_NAME
 LC_ADDRESS
 postal_fmt    "%f%N%a%N%d%N%b%N%s %h %e %r%N%z %T%N%c%N"
 % TODO
-country_name "Turkey"
+country_name "Tirkiye"
 country_post "TR"
 country_ab2  "TR"
 country_ab3  "TUR"
 country_num  792
 country_isbn 975
 country_car    "TR"
-lang_name   "kurdi"
+lang_name   "kurd<U00EE>"
 lang_ab      "ku"
 lang_term    "kur"
 lang_lib     "kur"
diff --git a/localedata/locales/mag_IN b/localedata/locales/mag_IN
index b8f65f70e26f4481..c86d9ba075ef2577 100644
--- a/localedata/locales/mag_IN
+++ b/localedata/locales/mag_IN
@@ -152,6 +152,7 @@ END LC_NAME
 LC_ADDRESS
 
 postal_fmt  "%z%c%T%s%b%e%r"
+country_name "<U092D><U093E><U0930><U0924>"
 country_ab2 "IN"
 country_ab3 "IND"
 country_num 356
diff --git a/localedata/locales/mfe_MU b/localedata/locales/mfe_MU
index 7415e9250fb78d50..7fabc5f80d9ae5e5 100644
--- a/localedata/locales/mfe_MU
+++ b/localedata/locales/mfe_MU
@@ -170,7 +170,7 @@ END LC_NAME
 
 LC_ADDRESS
 postal_fmt   "%f%N%h%s%N%T"
-country_name "Mauritius"
+country_name "Moris"
 country_ab2  "MU"
 country_ab3  "MUS"
 country_num  480
diff --git a/localedata/locales/mhr_RU b/localedata/locales/mhr_RU
index 85ac21b35af25c87..02bec3e9aed6fddc 100644
--- a/localedata/locales/mhr_RU
+++ b/localedata/locales/mhr_RU
@@ -159,6 +159,8 @@ country_ab2 "RU"
 country_ab3 "RUS"
 country_num 643
 country_car    "RUS"
+% марий йылме
+lang_name    "<U043C><U0430><U0440><U0438><U0439> <U0439><U044B><U043B><U043C><U0435>"
 lang_term    "mhr"
 lang_lib    "mhr"
 END LC_ADDRESS
diff --git a/localedata/locales/mi_NZ b/localedata/locales/mi_NZ
index 782f02359cb096d4..d08bc249427581c2 100644
--- a/localedata/locales/mi_NZ
+++ b/localedata/locales/mi_NZ
@@ -157,7 +157,7 @@ country_ab2   "NZ"
 country_ab3   "NZL"
 country_num   554
 country_car   "NZ"
-lang_name    "Te Reo"
+lang_name    "M<U0101>ori"
 lang_ab      "mi"
 lang_term    "mri"
 lang_lib    "mao"
diff --git a/localedata/locales/ms_MY b/localedata/locales/ms_MY
index 66b5dd98e99e5da3..8ffe9156b6e3d022 100644
--- a/localedata/locales/ms_MY
+++ b/localedata/locales/ms_MY
@@ -184,7 +184,7 @@ country_ab3 "MYS"
 country_num 458
 country_car "MAL"
 % Bahasa Melayu
-lang_name    "Bahasa Melayu"
+lang_name    "Melayu"
 lang_ab      "ms"
 lang_term    "msa"
 lang_lib    "may"
diff --git a/localedata/locales/my_MM b/localedata/locales/my_MM
index d612e5305aff7e2b..846363223883723b 100644
--- a/localedata/locales/my_MM
+++ b/localedata/locales/my_MM
@@ -309,7 +309,8 @@ country_ab2     "MM"
 country_num 104
 country_car "MYA"
 lang_ab         "my"
-lang_name       "<U1017><U1019><U102C>"
+% မြန်မာ
+lang_name       "<U1019><U103C><U1014><U103A><U1019><U102C>"
 lang_term   "mya"
 lang_lib    "bur"
 END LC_ADDRESS
diff --git a/localedata/locales/nan_TW b/localedata/locales/nan_TW
index f5bc5d1642c2ce9b..53884e02ffd16d2c 100644
--- a/localedata/locales/nan_TW
+++ b/localedata/locales/nan_TW
@@ -188,8 +188,8 @@ country_ab3  "TWN"
 country_num  158
 country_car "RC"
 country_isbn 957
-% 漢語閩南語
-lang_name    "<U6F22><U8A9E><U95A9><U5357><U8A9E>"
+% 閩南語
+lang_name    "<U95A9><U5357><U8A9E>"
 lang_term    "nan"
 lang_lib     "nan"
 END LC_ADDRESS
diff --git a/localedata/locales/nan_TW@latin b/localedata/locales/nan_TW@latin
index d4579a4cdf032d4c..104f4152a98b37ab 100644
--- a/localedata/locales/nan_TW@latin
+++ b/localedata/locales/nan_TW@latin
@@ -165,8 +165,8 @@ country_ab3	"TWN"
 country_num	158
 country_car "RC"
 country_isbn	957
-%lang_name "Bân-lâm-gú, Hō-ló-oē"
-lang_name "B<U00E2>n-l<U00E2>m-g<U00FA>, H<U014D>-l<U00F3>-o<U0113>"
+%lang_name "Bân-lâm-gú"
+lang_name "B<U00E2>n-l<U00E2>m-g<U00FA>"
 lang_term "nan"
 lang_lib "nan"
 END LC_ADDRESS
diff --git a/localedata/locales/nds_DE b/localedata/locales/nds_DE
index 232f9e4455ca74cc..d6200b7972b6b820 100644
--- a/localedata/locales/nds_DE
+++ b/localedata/locales/nds_DE
@@ -45,7 +45,7 @@ country_ab3   "DEU"
 country_car   "D"
 country_num 276
 country_isbn "3"
-lang_name "Neddersassisch"
+lang_name "Neddersass<U2019>sch"
 %lang_ab
 lang_term "nds"
 lang_lib "nds"
diff --git a/localedata/locales/nds_NL b/localedata/locales/nds_NL
index 8c8bebd93530938f..9fd50fb6256d3823 100644
--- a/localedata/locales/nds_NL
+++ b/localedata/locales/nds_NL
@@ -44,7 +44,7 @@ country_ab3 "NLD"
 country_car "NL"
 country_num 528
 country_isbn "3"
-lang_name "Neddersassisch"
+lang_name "Neddersass<U2019>sch"
 %lang_ab
 lang_term "nds"
 lang_lib "nds"
diff --git a/localedata/locales/nhn_MX b/localedata/locales/nhn_MX
index 88a89765e88f060b..2dbfcb3c6572cd0c 100644
--- a/localedata/locales/nhn_MX
+++ b/localedata/locales/nhn_MX
@@ -133,11 +133,12 @@ END LC_NAME
 
 LC_ADDRESS
 postal_fmt    "%f%N%a%N%d%N%b%N%s %h %e %r%N%z %T%N%c%N"
+country_name  "Mexihco"
 country_ab2   "MX"
 country_ab3   "MEX"
 country_num   484
 country_car    "MEX"
-lang_name    "Tlaxcala-Puebla Nahuatl"
+lang_name    "Tlahco n<U0101>huatlaht<U014D>lli"
 lang_term    "nhn"
 lang_lib    "nhn"
 END LC_ADDRESS
diff --git a/localedata/locales/niu_NU b/localedata/locales/niu_NU
index 553c5d9edcddfc86..387e109109fc0374 100644
--- a/localedata/locales/niu_NU
+++ b/localedata/locales/niu_NU
@@ -163,10 +163,12 @@ END LC_NAME
 
 LC_ADDRESS
 postal_fmt   "%f%N%a%N%d%N%b%N%s %h %e %r%N%z %T%N%c%N"
+country_name "Niu<U0113>"
 country_post "NU"
 country_ab2  "NU"
 country_ab3  "NIU"
 country_num  570
+lang_name    "ko e vagahau Niu<U0113>"
 lang_term    "niu"
 lang_lib    "niu"
 END LC_ADDRESS
diff --git a/localedata/locales/niu_NZ b/localedata/locales/niu_NZ
index 560101b447c22251..2eba7d2630f8a46d 100644
--- a/localedata/locales/niu_NZ
+++ b/localedata/locales/niu_NZ
@@ -99,6 +99,7 @@ country_ab2  "NZ"
 country_ab3  "NZL"
 country_num  554
 country_car    "NZ"
+lang_name    "ko e vagahau Niu<U0113>"
 lang_term    "niu"
 lang_lib    "niu"
 END LC_ADDRESS
diff --git a/localedata/locales/nr_ZA b/localedata/locales/nr_ZA
index 7de6420a6b52977e..f64434fadbad4e69 100644
--- a/localedata/locales/nr_ZA
+++ b/localedata/locales/nr_ZA
@@ -225,7 +225,7 @@ country_car   "ZA"
 country_num 710
 
 % Language name in Southern Ndebele
-lang_name "IsiNdebele"
+lang_name "isiNdebele"
 
 % ISO 639 two and three letter language names
 % see http://www.loc.gov/standards/iso639-2/englangn.html
diff --git a/localedata/locales/oc_FR b/localedata/locales/oc_FR
index 707927ee2662ac91..a106f8d1c1152959 100644
--- a/localedata/locales/oc_FR
+++ b/localedata/locales/oc_FR
@@ -44,7 +44,7 @@ country_ab3 "FRA"
 country_num 250
 country_isbn "979-10"
 country_car "F"
-lang_name    "Occitan"
+lang_name    "occitan"
 lang_ab      "oc"
 lang_term    "oci"
 lang_lib     "oci"
diff --git a/localedata/locales/or_IN b/localedata/locales/or_IN
index ef28b58895c342b6..2e4975e7b710b592 100644
--- a/localedata/locales/or_IN
+++ b/localedata/locales/or_IN
@@ -20,7 +20,7 @@ contact     ""
 email       "bug-glibc@gnu.org"
 tel         ""
 fax         ""
-language    "Oriya"
+language    "Odia"
 territory   "India"
 revision    "1.0"
 date        "2006-05-25"
diff --git a/localedata/locales/pa_PK b/localedata/locales/pa_PK
index 1f49bdc90d7ce637..0a584114b8f06590 100644
--- a/localedata/locales/pa_PK
+++ b/localedata/locales/pa_PK
@@ -173,7 +173,7 @@ END LC_NAME
 LC_ADDRESS
 % FIXME
 postal_fmt    "%a%N%f%N%d%N%b%N%h %s %e %r%N%T %z%N%c%N"
-country_name "<U0A2A><U0A3E><U0A15><U0A3F><U0A38><U0A24><U0A3E><U0A28>"
+country_name "<U067E><U06A9><U0633><U062A><U0627><U0646>"
 country_ab2 "PK"
 country_ab3 "PAK"
 country_num 586
diff --git a/localedata/locales/ps_AF b/localedata/locales/ps_AF
index 66f560ef4406be61..23dc86dcf1c1453b 100644
--- a/localedata/locales/ps_AF
+++ b/localedata/locales/ps_AF
@@ -308,8 +308,7 @@ END LC_NAME
 
 LC_ADDRESS
 postal_fmt	"%a%N%f%N%d%N%b%N%s %h %e %r%N%C-%z %T%N%c%N"
-country_name	"<U0627><U0641><U063A><U0627><U0646>/
-<U0633><U062A><U0627><U0646>"
+country_name	"<U0627><U0641><U063A><U0627><U0646><U0633><U062A><U0627><U0646>"
 country_post	"AF"
 country_ab2	"AF"
 country_ab3	"AFG"
diff --git a/localedata/locales/quz_PE b/localedata/locales/quz_PE
index f6b1956b937100df..9ed890cbb0e6c762 100644
--- a/localedata/locales/quz_PE
+++ b/localedata/locales/quz_PE
@@ -10,7 +10,7 @@ escape_char /
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %
-% Quechua (Cusco-Collao) language locale for Peru
+% Cusco Quechua language locale for Peru
 %
 % Prepared and contributed to glibc by Chris Leonard <cjl@sugarlabs.org>
 % and Amos Batto
@@ -25,14 +25,14 @@ escape_char /
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
 LC_IDENTIFICATION
-title        "Quechua (Cusco-Collao) locale for Peru"
+title        "Cusco Quechua locale for Peru"
 source       "Sugar Labs // OLPC"
 address      ""
 contact      "sugarlabs.org"
 email        "libc-alpha@sourceware.org"
 tel          ""
 fax          ""
-language     "Quechua (Cusco-Collao)"
+language     "Cusco Quechua"
 territory    "Peru"
 revision     "1.0"
 date         "2013-08-24"
@@ -144,7 +144,7 @@ country_ab2   "PE"
 country_ab3   "PER"
 country_num   604
 country_car   "PE"
-lang_name    "Quechua (Cusco)"
+lang_name    "Qusqu runasimi"
 lang_term    "quz"
 lang_lib    "quz"
 END LC_ADDRESS
diff --git a/localedata/locales/raj_IN b/localedata/locales/raj_IN
index ece080223e8de94b..ce009e406ab5c332 100644
--- a/localedata/locales/raj_IN
+++ b/localedata/locales/raj_IN
@@ -158,7 +158,16 @@ name_ms     "<U0915><U0941><U092E><U093E><U0930>"
 END LC_NAME
 
 LC_ADDRESS
-copy "hi_IN"
+postal_fmt  "%z%c%T%s%b%e%r"
+country_name "<U092D><U093E><U0930><U0924>"
+country_ab2 "IN"
+country_ab3 "IND"
+country_num 356
+country_car "IND"
+% राजस्थानी
+lang_name    "<U0930><U093E><U091C><U0938><U094D><U0925><U093E><U0928><U0940>"
+lang_term    "raj"
+lang_lib    "raj"
 END LC_ADDRESS
 
 LC_TELEPHONE
diff --git a/localedata/locales/rw_RW b/localedata/locales/rw_RW
index e0bc763c5a484c15..31cb4673dfa4a3aa 100644
--- a/localedata/locales/rw_RW
+++ b/localedata/locales/rw_RW
@@ -132,7 +132,7 @@ END LC_NAME
 
 LC_ADDRESS
 postal_fmt    "%f%N%a%N%d%N%b%N%s %h %e %r%N%z %T%N%c%N"
-country_name   "Rwanda"
+country_name   "U Rwanda"
 country_ab2    "RW"
 country_ab3    "RWA"
 country_num 646
diff --git a/localedata/locales/sah_RU b/localedata/locales/sah_RU
index 80f67219c39cceb2..2b5a4b7bd660fb6e 100644
--- a/localedata/locales/sah_RU
+++ b/localedata/locales/sah_RU
@@ -1,7 +1,7 @@
 escape_char  /
 comment_char  %
 
-% Yakut (Sakha) locale for Russian Federation
+% Sakha (Yakut) locale for Russian Federation
 % Source: Valery Timiriliyev
 % Email: timiriliyev@gmail.com
 % Tel:
@@ -13,14 +13,14 @@ comment_char  %
 % Users: general
 
 LC_IDENTIFICATION
-title      "Yakut (Sakha) locale for Russian Federation"
+title      "Sakha (Yakut) locale for Russian Federation"
 source     "Valery Timiriliyev"
 address    ""
 contact    "Valery Timiriliyev"
 email      "timiriliyev@gmail.com"
 tel        ""
 fax        ""
-language   "Yakut"
+language   "Sakha"
 territory  "Russian Federation"
 revision   "1.1.0"
 date       "2018-07-06"
@@ -264,11 +264,11 @@ END LC_NAME
 LC_ADDRESS
 postal_fmt    "%f%N%a%N%d%N%b%N%s %h %e %r%N%z %T%N%c%N"
 
-% Россия
-country_name  "<U0420><U043E><U0441><U0441><U0438><U044F>"
+% Арассыыйа
+country_name  "<U0410><U0440><U0430><U0441><U0441><U044B><U044B><U0439><U0430>"
 
-% Саха тыла
-lang_name     "<U0421><U0430><U0445><U0430> <U0442><U044B><U043B><U0430>"
+% саха тыла
+lang_name     "<U0441><U0430><U0445><U0430><U0020><U0442><U044B><U043B><U0430>"
 
 % UN Geneve 1949:68 Distinguishing signs of vehicles in international traffic
 % RUS
diff --git a/localedata/locales/sat_IN b/localedata/locales/sat_IN
index 134eb8a572c836a7..e1775a4072ff7d9c 100644
--- a/localedata/locales/sat_IN
+++ b/localedata/locales/sat_IN
@@ -158,12 +158,13 @@ END LC_NAME
 LC_ADDRESS
 
 postal_fmt  "%z%c%T%s%b%e%r"
+country_name "<U1C65><U1C64><U1C67><U1C5A><U1C5B>"
 country_ab2 "IN"
 country_ab3 "IND"
 country_num 356
 country_car    "IND"
-% Satār
-lang_name    "Sat<U0101>r"
+% ᱥᱟᱱᱛᱟᱲᱤ
+lang_name    "<U1C65><U1C5F><U1C71><U1C5B><U1C5F><U1C72><U1C64>"
 lang_term    "sat"
 lang_lib    "sat"
 END LC_ADDRESS
diff --git a/localedata/locales/sc_IT b/localedata/locales/sc_IT
index 1488744575dae32a..32626404d2956e8c 100644
--- a/localedata/locales/sc_IT
+++ b/localedata/locales/sc_IT
@@ -144,7 +144,7 @@ country_ab3 "ITA"
 country_num 380
 country_isbn "978-88,979-12"
 country_car  "I"
-lang_name    "Sardu"
+lang_name    "sardu"
 lang_ab      "sc"
 lang_term    "srd"
 lang_lib    "srd"
diff --git a/localedata/locales/sd_IN b/localedata/locales/sd_IN
index 66aa0e254cefe0d7..64c5366c31dcd818 100644
--- a/localedata/locales/sd_IN
+++ b/localedata/locales/sd_IN
@@ -164,12 +164,14 @@ LC_ADDRESS
 % This is the ISO_IEC TR14652 Locale definition for the LC_ADDRESS category
 % generated by IBM Basic CountryPack Transformer.
 postal_fmt  "%z%c%T%s%b%e%r"
-% https://sd.wikipedia.org/wiki/%DA%80%D8%A7%D8%B1%D8%AA : "ڀارت"
-country_name "<U0680><U0627><U0631><U062A>"
+% From cldr: انڊيا
+country_name "<U0627><U0646><U068A><U064A><U0627>"
 country_ab2 "IN"
 country_ab3 "IND"
 country_num 356
 country_car    "IND"
+% سنڌي
+lang_name    "<U0633><U0646><U068C><U064A>"
 lang_ab      "sd"
 lang_term    "snd"
 lang_lib    "snd"
diff --git a/localedata/locales/sd_IN@devanagari b/localedata/locales/sd_IN@devanagari
index b1ce87df93f9c3d6..97357e08559ea64f 100644
--- a/localedata/locales/sd_IN@devanagari
+++ b/localedata/locales/sd_IN@devanagari
@@ -167,6 +167,8 @@ country_ab2 "IN"
 country_ab3 "IND"
 country_num 356
 country_car    "IND"
+% सिन्धी
+lang_name    "<U0938><U093F><U0928><U094D><U0927><U0940>"
 lang_ab      "sd"
 lang_term    "snd"
 lang_lib    "snd"
diff --git a/localedata/locales/shn_MM b/localedata/locales/shn_MM
index 4212c50ec5ae0066..a837f5fb394b1bb5 100644
--- a/localedata/locales/shn_MM
+++ b/localedata/locales/shn_MM
@@ -281,7 +281,8 @@ END LC_NAME
 
 LC_ADDRESS
 postal_fmt   "%a%N%d%N%f%N%b%N%h%t%r%t%e%t%s%N%T%t%z%N%c%N"
-country_name "<U1019><U1062><U107C><U103A><U1088>"
+% https://shn.wikipedia.org/wiki/%E1%80%99%E1%80%AD%E1%80%B0%E1%80%84%E1%80%BA%E1%80%B8%E1%80%99%E1%81%A2%E1%81%BC%E1%80%BA%E1%82%88 မိူင်းမၢၼ်ႈ
+country_name "<U1019><U102D><U1030><U1004><U103A><U1038><U1019><U1062><U107C><U103A><U1088>"
 country_post "Myanmar"
 country_ab2  "MM"
 country_num  104
diff --git a/localedata/locales/shs_CA b/localedata/locales/shs_CA
index a5b675a316945608..ab48d600ef72ba75 100644
--- a/localedata/locales/shs_CA
+++ b/localedata/locales/shs_CA
@@ -8,7 +8,7 @@ escape_char /
 % exempt you from the conditions of the license if your use would
 % otherwise be governed by that license.
 
-% Secwepemctsin (Shuswap) language locale for Canada
+% Shuswap language locale for Canada
 % sorting according to CAN/CSA-Z243.4.1-1992
 % Source: Neskie Manuel
 % Address: 745 Ska-Hiish Dr
@@ -25,14 +25,14 @@ escape_char /
 % Users: general
 
 LC_IDENTIFICATION
-title      "Secwepemctsin locale for Canada"
+title      "Shuswap locale for Canada"
 source     "Neskie Manuel"
 address    "745 Ska-Hiish Dr, Chase BC V0E 1M3"
 contact    ""
 email      "bug-glibc-locales@gnu.org"
 tel        ""
 fax        ""
-language   "Secwepemctsin"
+language   "Shuswap"
 territory  "Canada"
 revision   "1.0"
 date       "2008-01-15"
@@ -155,7 +155,7 @@ country_ab2 "CA"
 country_ab3 "CAN"
 country_num 124
 country_car    "CDN"
-lang_name "Secwepemctsin"
+lang_name "Secwepemcts<U00ED>n"
 lang_term "shs"
 lang_lib "shs"
 END LC_ADDRESS
diff --git a/localedata/locales/sm_WS b/localedata/locales/sm_WS
index 6058fbdc38283bb2..2823005c0635b6cf 100644
--- a/localedata/locales/sm_WS
+++ b/localedata/locales/sm_WS
@@ -159,7 +159,7 @@ END LC_NAME
 
 LC_ADDRESS
 postal_fmt    "%a%N%f%N%d%N%b%N%h %s %e %r%N%T, %S %z%N%c%N"
-country_name "Samoa"
+country_name "S<U0101>moa"
 country_post "%a %N   %f %N   %d %N   %b %N   %h %s %e %/
 r %N   %T, %c  %N"
 % http://laendercode.net/en/2-letter-list.html
diff --git a/localedata/locales/ss_ZA b/localedata/locales/ss_ZA
index 7532a1940b3556f9..8e08a85f461aed9b 100644
--- a/localedata/locales/ss_ZA
+++ b/localedata/locales/ss_ZA
@@ -228,7 +228,7 @@ country_car   "ZA"
 % country_isbn ""
 
 % Language name in Swati
-lang_name "SiSwati"
+lang_name "siSwati"
 
 % ISO 639 two and three letter language names
 % see http://www.loc.gov/standards/iso639-2/englangn.html
diff --git a/localedata/locales/szl_PL b/localedata/locales/szl_PL
index 8d5de2112e54b7ad..88f1df4e905c5bdf 100644
--- a/localedata/locales/szl_PL
+++ b/localedata/locales/szl_PL
@@ -188,6 +188,7 @@ END LC_NAME
 
 LC_ADDRESS
 postal_fmt    "%f%N%a%N%d%N%b%N%s %h %e %r%N%z %T%N%c%N"
+country_name "Polska"
 country_ab2 "PL"
 country_ab3 "POL"
 country_num 616
diff --git a/localedata/locales/te_IN b/localedata/locales/te_IN
index fb5bb21b09ae0647..cb1b1109690b20bb 100644
--- a/localedata/locales/te_IN
+++ b/localedata/locales/te_IN
@@ -177,7 +177,7 @@ LC_ADDRESS
 % generated by IBM Basic CountryPack Transformer.
 
 postal_fmt      "%a%N%d%N%f%N%r%t%e%N%h%t%b%N%s%t%N%T%t%z%N%S%N%c"
-country_name    "<U0C2D><U0C3E><U0C30><U0C24> <U0C26><U0C47><U0C36><U0C02>"
+country_name    "<U0C2D><U0C3E><U0C30><U0C24><U0C26><U0C47><U0C36><U0C02>"
 country_ab2     "IN"
 country_ab3     "IND"
 country_num     356
diff --git a/localedata/locales/tg_TJ b/localedata/locales/tg_TJ
index 35dfca4574d4b5b9..6d1e4d8ab339501f 100644
--- a/localedata/locales/tg_TJ
+++ b/localedata/locales/tg_TJ
@@ -222,6 +222,8 @@ country_ab2 "TJ"
 country_ab3 "TJK"
 country_num 762
 country_car    "TJ"
+% тоҷикӣ
+lang_name    "<U0442><U043E><U04B7><U0438><U043A><U04E3>"
 lang_ab      "tg"
 lang_term    "tgk"
 lang_lib    "tgk"
diff --git a/localedata/locales/the_NP b/localedata/locales/the_NP
index 993e62fbfc5cea64..296e9970885ee490 100644
--- a/localedata/locales/the_NP
+++ b/localedata/locales/the_NP
@@ -162,10 +162,14 @@ LC_ADDRESS
 postal_fmt  "%z%c%T%s%b%e%r"
 lang_term    "the"
 lang_lib     "the"
+% नेपाल
+country_name "<U0928><U0947><U092A><U093E><U0932>"
 country_ab2 "NP"
 country_ab3 "NPL"
 country_num 524
 country_car "NEP"
+% थारु
+lang_name   "<U0925><U093E><U0930><U0941>"
 END LC_ADDRESS
 
 
diff --git a/localedata/locales/tk_TM b/localedata/locales/tk_TM
index 410afaf6c76c920b..e29ae20408763196 100644
--- a/localedata/locales/tk_TM
+++ b/localedata/locales/tk_TM
@@ -399,7 +399,7 @@ country_num  795
 country_ab2 "TM"
 country_ab3 "TKM"
 % Türkmençe
-lang_name   "t<U00FC>rkmen<U00E7>e"
+lang_name   "t<U00FC>rkmen dili"
 lang_term   "tuk"
 lang_lib    "tuk"
 lang_ab     "tk"
diff --git a/localedata/locales/tl_PH b/localedata/locales/tl_PH
index 40fd71d96095be12..03b8350749ab2c0a 100644
--- a/localedata/locales/tl_PH
+++ b/localedata/locales/tl_PH
@@ -138,6 +138,7 @@ country_ab2 "PH"
 country_ab3 "PHL"
 country_num 608
 country_car    "RP"
+lang_name    "Tagalog"
 lang_ab      "tl"
 lang_term    "tgl"
 lang_lib    "tgl"
diff --git a/localedata/locales/to_TO b/localedata/locales/to_TO
index 7abe8685df8488cf..403a1219126bb189 100644
--- a/localedata/locales/to_TO
+++ b/localedata/locales/to_TO
@@ -169,7 +169,7 @@ country_ab3  "TON"
 country_car  "TON"
 country_num 776
 % Tongan
-lang_name    "Tonga<U000A>"
+lang_name    "lea fakatonga"
 % https://en.wikipedia.org/wiki/Tongan_language
 lang_ab      "to"
 lang_term    "ton"
diff --git a/localedata/locales/tpi_PG b/localedata/locales/tpi_PG
index 3315c27633ba58c6..885481974c8b531b 100644
--- a/localedata/locales/tpi_PG
+++ b/localedata/locales/tpi_PG
@@ -172,7 +172,8 @@ END LC_NAME
 LC_ADDRESS
 % http://www.addressexamples.com/papua-new-guinea-address-format/
 postal_fmt    "%a%s%z%C"
-country_name "Papua New Guinea"
+% https://tpi.wikipedia.org/wiki/Papua_Niugini
+country_name "Papua Niugini"
 country_post ""
 country_ab2  "PG"
 country_ab3  "PNG"
@@ -180,7 +181,7 @@ country_car  "PNG"
 % https://en.wikipedia.org/wiki/ISO_3166-1_numeric
 country_num 598
 % Tok Pisin
-lang_name    "Tok  Pisin"
+lang_name    "Tok Pisin"
 % https://en.wikipedia.org/wiki/Tok_Pisin
 lang_ab      ""
 lang_term    "tpi"
diff --git a/localedata/locales/tt_RU@iqtelif b/localedata/locales/tt_RU@iqtelif
index d4737c888e41352b..b52b1299946f372f 100644
--- a/localedata/locales/tt_RU@iqtelif
+++ b/localedata/locales/tt_RU@iqtelif
@@ -155,10 +155,12 @@ END LC_MEASUREMENT
 
 LC_ADDRESS
 postal_fmt    "%f%N%a%N%d%N%b%N%s %h %e %r%N%z %T%N%c%N"
+country_name  "Ur<U0131>s Pat<U015F>ahl<U0131>q"
 country_ab2 "RU"
 country_ab3 "RUS"
 country_num 643
 country_car    "RUS"
+lang_name    "tatar tele"
 lang_ab      "tt"
 lang_term    "tat"
 lang_lib    "tat"
diff --git a/localedata/locales/ug_CN b/localedata/locales/ug_CN
index 1ba583c588e074d2..acb8b68239c4a653 100644
--- a/localedata/locales/ug_CN
+++ b/localedata/locales/ug_CN
@@ -195,6 +195,8 @@ country_ab3	"CHN"
 country_num	156
 country_car "CHN"
 country_isbn	7
+% ئۇيغۇرچە
+lang_name    "<U0626><U06C7><U064A><U063A><U06C7><U0631><U0686><U06D5>"
 lang_ab      "ug"
 lang_term    "uig"
 lang_lib    "uig"
diff --git a/localedata/locales/unm_US b/localedata/locales/unm_US
index 1e62c60443b83ec6..ef458e22dc195809 100644
--- a/localedata/locales/unm_US
+++ b/localedata/locales/unm_US
@@ -139,7 +139,7 @@ country_ab3   "USA"
 country_num   840
 country_car   "USA"
 country_isbn  0
-% lang_name     ""
+lang_name     "Lenape"
 % lang_ab       ""
 lang_term    "unm"
 lang_lib    "unm"
diff --git a/localedata/locales/wa_BE b/localedata/locales/wa_BE
index e97493089eb287b0..afec10f41c8b1612 100644
--- a/localedata/locales/wa_BE
+++ b/localedata/locales/wa_BE
@@ -44,7 +44,7 @@ country_ab3 "BEL"
 country_num 056
 country_isbn "2"
 country_car  "B"
-lang_name "Walon"
+lang_name "walon"
 lang_ab      "wa"
 lang_term    "wln"
 lang_lib     "wln"
diff --git a/localedata/locales/wo_SN b/localedata/locales/wo_SN
index 47263d2eab6db94e..20190925034c9ab4 100644
--- a/localedata/locales/wo_SN
+++ b/localedata/locales/wo_SN
@@ -161,6 +161,7 @@ country_ab2 "SN"
 country_ab3 "SEN"
 country_num 686
 country_car "SN"
+lang_name    "Wolof"
 lang_ab      "wo"
 lang_term    "wol"
 lang_lib    "wol"
diff --git a/localedata/locales/xh_ZA b/localedata/locales/xh_ZA
index 4564137e851b0cda..327f7250987aaa57 100644
--- a/localedata/locales/xh_ZA
+++ b/localedata/locales/xh_ZA
@@ -209,13 +209,13 @@ END LC_NAME
 
 LC_ADDRESS
 % https://xh.wikipedia.org/wiki/UMzantsi_Afrika
-country_name "UMzantsi Afrika"
+country_name "uMzantsi Afrika"
 %
 % Abbreviated country postal name
 country_post "ZA"
 %
 % Language name in Sotho
-lang_name "IsiXhosa"
+lang_name "isiXhosa"
 
 % UN Geneve 1949:68 Distinguishing signs of vehicles in international traffic
 % http://www.unece.org/trans/conventn/disting-signs-5-2001.pdf
diff --git a/localedata/locales/yo_NG b/localedata/locales/yo_NG
index ec72d078bb7eab6d..a3102270bf0fd38b 100644
--- a/localedata/locales/yo_NG
+++ b/localedata/locales/yo_NG
@@ -271,8 +271,8 @@ LC_ADDRESS
 % "country designation for the <country_post> keyword",
 % "end of line
 postal_fmt    "%f%N%a%N%d%N%b%N%s %h %e %r%N%z %T%N%c%N"
-
-country_name "Or<U00ED>l<U1EB9><U0301><U00E8>de N<U00E0><U00EC>j<U00ED>r<U00ED><U00E0>"
+% https://yo.wikipedia.org/wiki/N%C3%A0%C3%ACj%C3%ADr%C3%AD%C3%A0 and CLDR: Orilẹ̀-èdè Nàìjíríà
+country_name "Oril<U1EB9><U0300><U002D><U00E8>d<U00E8> N<U00E0><U00EC>j<U00ED>r<U00ED><U00E0>"
 
 % Language name in Yoruba - "Yorùbá"
 lang_name     "<U00C8>d<U00E8> Yor<U00F9>b<U00E1>"
diff --git a/localedata/locales/yuw_PG b/localedata/locales/yuw_PG
index 0cb3cadf4ae485a1..55e787a5cdfc559f 100644
--- a/localedata/locales/yuw_PG
+++ b/localedata/locales/yuw_PG
@@ -16,7 +16,7 @@ contact    "Hannah Sarvasy"
 email      "nungon.localization@gmail.com"
 tel        ""
 fax        ""
-language   "Yau/Nungon"
+language   "Yau"
 territory  "Papua New Guinea"
 revision   "1.0"
 date       "2016-12-07"
@@ -133,8 +133,8 @@ country_ab2  "PG"
 country_ab3  "PNG"
 % ISO 3166-1 numeric code for PNG
 country_num 598
-% Yau/Nungon
-lang_name    "Yau/Nungon"
+% See: https://en.wikipedia.org/wiki/Yau_language, the endonym seems to be Uruwa
+lang_name    "Uruwa"
 country_car  "PNG"
 lang_ab      ""
 lang_term    "yuw"
diff --git a/localedata/locales/zh_HK b/localedata/locales/zh_HK
index c130878f3d6e0ace..e8097d6ed0704670 100644
--- a/localedata/locales/zh_HK
+++ b/localedata/locales/zh_HK
@@ -178,6 +178,7 @@ country_ab2 "HK"
 country_ab3 "HKG"
 country_num 344
 country_car "HK"
+lang_name       "<U7E41><U9AD4><U4E2D><U6587>"
 lang_ab		"zh"
 lang_term	"zho"
 lang_lib	"chi"
diff --git a/localedata/locales/zh_SG b/localedata/locales/zh_SG
index 2427cd3ead3583dc..472843c7f77b8ed6 100644
--- a/localedata/locales/zh_SG
+++ b/localedata/locales/zh_SG
@@ -175,6 +175,7 @@ country_ab3 "SGP"
 country_num 702
 % SGP
 country_car    "SGP"
+lang_name       "<U7B80><U4F53><U4E2D><U6587>"
 lang_ab		"zh"
 lang_term	"zho"
 lang_lib	"chi"
